nearlib/account.js

```diff
@@ -1,39 +1,85 @@
+const superagent = require('superagent');
+const ed25519 = require('ed25519');
+const bs58 = require('bs58');
+const MAX_RETRIES = 3;
 
-/**
- * Creates a new account with a random new key.
- */
-exports.createRandomAccount = async function() {
-    // Not implemented yet
-};
+class Account {
+    constructor(keyStore) {
```

Comment:
Not sure why `Account` should know about key store vs passing key directly.  I can see it being useful if e.g. signing happens on keystore level so that there is no need to read key here.
In that case key store can e.g. be backed by hardware device doing the signing without key bytes ever leaving device.

@ilblackdragon what do you think?

Reply:
I could go with SignatureProvider -- good point about hardware-based devices.

Reply:
Let's not get hang up on this right now though, this can be easily refactored in the future now that we have tests.